hudi-common/src/main/java/org/apache/hudi/common/table/read/FileGroupRecordBuffer.java

```diff
@@ -349,11 +385,13 @@ protected boolean hasNextLogRecord() {
     while (logRecordIterator.hasNext()) {
       BufferedRecord<T> nextRecordInfo = logRecordIterator.next();
       if (!nextRecordInfo.isDelete()) {
-        nextRecord = nextRecordInfo.getRecord();
+        BufferedRecord<T> convertedBufferedRecord = applyOutputSchemaConversion(nextRecordInfo);
+        nextRecord = convertedBufferedRecord.getRecord();
+        handleBaseFileInsert(nextRecordInfo.getRecordKey(), nextRecord);
         readStats.incrementNumInserts();
         return true;
       } else if (emitDelete) {
-        nextRecord = readerContext.getDeleteRow(nextRecordInfo.getRecord(), nextRecordInfo.getRecordKey());
+        nextRecord = applyOutputSchemaConversion(readerContext.getDeleteRow(nextRecordInfo.getRecord(), nextRecordInfo.getRecordKey()));
```

Comment:
We just igore the deletes from logs, is this behavior in line with before? Ususally these deletes are invalid or probably because of disorder inputs from Network shufflle, not sure if use still want to see these changes to downstream.

Reply:
This matches the current logic [here](https://github.com/apache/hudi/blob/master/hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandleWithChangeLog.java#L106). It seems like the changes within the log files are not output to the CDCLogger, only the changes to the base file.